### PR TITLE
Fix error caused by loading string constants

### PR DESCRIPTION
**Pull request by ShimShtein**

### CODE REVIEW

File: lib/foreman_hooks/as_dependencies_hook.rb [hunk 1]
```diff
@@ -2,6 +2,7 @@ module ForemanHooks
   module ASDependenciesHook
     def load_missing_constant(from_mod, constant_name)
       super(from_mod, constant_name).tap do |ret|
+        return unless ret.try(:name)
```
Comment (adamruzicka) [inline lib/foreman_hooks/as_dependencies_hook.rb:5]: We should keep the interface and return the original return value.
```suggestion
        return ret unless ret.try(:name)
```

Comment (adamruzicka) [reply lib/foreman_hooks/as_dependencies_hook.rb:5]: Also can the autoloaded constant be something that responds to `name`, but returns a falsey value?

Comment (ShimShtein) [reply lib/foreman_hooks/as_dependencies_hook.rb]: Unless `klass` can be a falsey value, we're good.
I don't think it's possible at all.

**Comment by ShimShtein** [comment]
@adamruzicka, fixed, thanks!

**Review by adamruzicka — APPROVED** [review]
LGTM

**Comment by lzap** [comment]
Adam asked me to take a look, /CC

**Comment by lzap** [comment]
Sorry busy with my talk, tomorrow hopefully.

**Review by lzap — APPROVED** [review]
I have no context, there is no ticket nor description. However the code looks good in a sense.

**Comment by lzap** [comment]
Relevant BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1894009

Do you want me to cut a new release with this patch?

**Comment by adamruzicka** [comment]
> Do you want me to cut a new release with this patch?

Yes please

**Comment by lzap** [comment]
* https://github.com/theforeman/foreman-packaging/pull/6007
* https://github.com/theforeman/foreman-packaging/pull/6008

Downstream we can just pull this version, the plugin is not actively developed (no changes last 1.5 years).

**Pull request merged**